### DISCUSSION TITLE
Mains-powered device fast start

### DISCUSSION THREAD

**Discussion by flaviut** [arendst/Tasmota#13145] [Ideas]
When Tasmota is on a light switch, it is likely controlling mains-powered lights with bulk capacitors.

When the switch is toggled on, especially on minimum brightness, these capacitors take a few seconds to charge up.

It would be nice if Tasmota would allow for user-defined fast start in these cases. This could be as simple as "100% PWM for n milliseconds if it's been m seconds since the switch has been turned off", or as complicated as requiring the user to enter charge resistance, capacitance, and bleed resistance and using that to always have the perfect length of fast-start.

**Comment by sfromis** [comment]
Hardware-wise, Tasmota can't know when or for how long the device was powered off, and thus no option of implementing a "fast start" depending on such info. The general assumption for "smart" devices is that you do not turn them off (you lose full control via wifi), and instead wire the light switch to control an input pin, not power.

You can control boot time behavior using PowerOnState and rules. Using rules you could also check RestartReason (from the Info3 JSON payload) to suggest if it is a restart due to power.

**Reply by flaviut** [reply]
> Hardware-wise, Tasmota can't know when or for how long the device was powered off

Sure it does! When it's in a light switch, Tasmota is doing both the turning on and off & the ESP8266 has a built-in clock that is active even in sleep mode.

This is completely unnecessary when Tasmota is used in a bulb, the bulk capacitors are always charged at full current and the dimming is done with PWM on the DC.

**Reply by barbudor** [reply]
May be you can try using rules so on POWER ON you set dimmer at a suitable high level, start a RuleTimer and when the RuleTimer expires, set the expected dimming value
